lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -28,11 +28,16 @@
 class MultiGalaxyAPIProxy:
     """A proxy that abstracts talking to multiple Galaxy instances."""
 
-    def __init__(self, apis, concrete_artifacts_manager):
-        # type: (t.Iterable[GalaxyAPI], ConcreteArtifactsManager) -> None
+    def __init__(self, apis, concrete_artifacts_manager, offline=False):
+        # type: (t.Iterable[GalaxyAPI], ConcreteArtifactsManager, bool) -> None
         """Initialize the target APIs list."""
         self._apis = apis
         self._concrete_art_mgr = concrete_artifacts_manager
+        self._offline = offline  # Prevent all GalaxyAPI calls
+
+    def _offline_is_unsupported(self):
```

Comment:
This method name left me puzzled. As it is now, this looks like a property, functionally. That is, if you just look at the name. Then, looking at the invocations, it's confusing because it seems to do nothing important, but actually it interrupts the control flow in those places.

A better name could be something like
```suggestion
    def _assert_that_offline_is_supported(self):
```

Reply:
Or it could be
```suggestion
    def _raise_if_offline_is_unsupported(self):
```

Reply:
Thanks for the suggestion, I like `_assert_that_offline_mode_is_not_requested` too. Fixed.